test/e2e/stop_test.go

```diff
@@ -363,4 +364,47 @@ var _ = Describe("Podman stop", func() {
 		Expect(session).Should(Exit(0))
 		Expect(podmanTest.NumberOfContainersRunning()).To(Equal(0))
 	})
+
+	It("podman stop --filter", func() {
+		session1 := podmanTest.Podman([]string{"container", "create", ALPINE})
+		session1.WaitWithDefaultTimeout()
+		Expect(session1).Should(Exit(0))
+		cid1 := session1.OutputToString()
+
+		session2 := podmanTest.Podman([]string{"container", "create", ALPINE})
+		session2.WaitWithDefaultTimeout()
+		Expect(session2).Should(Exit(0))
+		cid2 := session2.OutputToString()
+		shortCid2 := cid2[0:5]
+
+		session3 := podmanTest.Podman([]string{"container", "create", ALPINE})
+		session3.WaitWithDefaultTimeout()
+		Expect(session3).Should(Exit(0))
+		cid3 := session3.OutputToString()
+		shortCid3 := cid3[0:5]
+
+		run1 := podmanTest.Podman([]string{"start", "--all"})
+		run1.WaitWithDefaultTimeout()
+		Expect(run1).Should(Exit(0))
+
+		session1 = podmanTest.Podman([]string{"stop", cid1, "-f", "status=running"})
+		session1.WaitWithDefaultTimeout()
+		Expect(session1).Should(Exit(0))
+		Expect(session1.OutputToString()).To(HaveLen(0))
+
+		session1 = podmanTest.Podman([]string{"stop", cid1, "-f", "status=exited"})
+		session1.WaitWithDefaultTimeout()
+		Expect(session1).Should(Exit(0))
+		Expect(session1.OutputToString()).To(BeEquivalentTo(cid1))
+
+		session1 = podmanTest.Podman([]string{"stop", cid2, "--filter", fmt.Sprintf("id=%s", shortCid2)})
+		session1.WaitWithDefaultTimeout()
+		Expect(session1).Should(Exit(0))
+		Expect(session1.OutputToString()).To(BeEquivalentTo(cid2))
```

Comment:
Would this test (and the one above) make more sense if there was an `--all` flag rather than explicitly stating something like `cid2`? Since by stating `cid2` guarantees it will stop the container?